### DISCUSSION TITLE
Flux.1 Tools released (Fill + Redux + Depth + Canny)

### DISCUSSION THREAD

**Discussion by GuidoBartoli** [lllyasviel/stable-diffusion-webui-forge#2357] [Ideas]
Black Forest Labs has just released **Flux.1 Tools**: https://blackforestlabs.ai/flux-1-tools/

- **FLUX.1 Fill**: State-of-the-art inpainting and outpainting models, enabling editing and expansion of real and generated images given a text description and a binary mask.
- **FLUX.1 Depth**: Models trained to enable structural guidance based on a depth map extracted from an input image and a text prompt.
- **FLUX.1 Canny**: Models trained to enable structural guidance based on canny edges extracted from an input image and a text prompt.
- **FLUX.1 Redux**: An adapter that allows mixing and recreating input images and text prompts.

(Flux.1 Dev variants of these tools are available on Hugging Face)

It seems the best solution to replace Flux ControlNets... hoping that we will see them supported in Stable Diffusion Forge soon! 😉

**Comment by StellarNear** [comment]
+1 need that in the best UI forge !

**Comment by dhosein** [comment]
Oh wow... I have soo much use for those tools. Hope they can be integrated.

**Reply by SunGreen777** [reply]
I would also like support for sd3 medium, They answered me on behalf of the developers, Tools requires code revision, it may take a year or maybe 1 day, that’s what they wrote.

**Comment by MigCurto** [comment]
I can see FluxTools Loras as a decent alternative to controlnet

**Comment by danividalg** [comment]
+1

**Comment by danilomaiaweb** [comment]
+1

**Comment by wzwow** [comment]
+100

doesnt IIIyasviel has a channel to accept donation? maybe that will motive him to continue update forge
